aten/src/ATen/native/native_functions.yaml

```diff
@@ -905,6 +909,11 @@
 - func: logsumexp_out(Tensor result, Tensor self, int64_t dim, bool keepdim=False) -> Tensor
   variants: function
 
+- func: masked_scale(Tensor self, Tensor mask, double scale) -> Tensor
```

Comment:
Hmm I'm not 100% sure if we want this in our API. Generally those cases should be served by `torch.where`. I know it's slightly less efficient, but in the end this should just be generated by the JIT. I guess we could have it as a private method (`_masked_scale`)

Reply:
Sure, will do `_masked_scale` and `_fused_dropout` - they'll still be available in python, right? I thought of using `where`, but it does not accept scalar argument as one of the tensors (it should, IMO, similar to comparison ops), and there's no way to scale in the same kernel, so it ends up being way slower. I totally agree this will be beside the point when randoms in jit are up and running, but until that, using more efficient dropout provides a few percent end-to-end speedup on some networks, so there's no reason not to use it.

Reply:
They will, they just no longer be `torch.masked_scale`, but `torch._C._masked_scale`. I agree about allowing `where` to broadcast scalars.